Sources/Core/JavaModelRenderer.swift

```diff
@@ -222,10 +222,25 @@ public struct JavaModelRenderer: JavaFileRenderer {
     }
 
     func renderBuilderBuild() -> JavaIR.Method {
-        let params = (transitiveProperties.map { param, _ in
+        let constructorParams = (transitiveProperties.map { param, _ in
             "this." + Languages.java.snakeCaseToPropertyName(param)
         } + ["this._bits"]).joined(separator: ",\n")
-        return JavaIR.method(annotations: [.nonnull], [.public], "\(className) build()") { ["return new " + self.className + "(", params, ");"] }
+
+        let missingFieldsCheck = !transitiveProperties.filter { $1.nullability == Nullability.nonnull }.isEmpty
+            ? ["List<String> missingFields = new ArrayList<>();"] + transitiveProperties.filter {
+                $1.nullability == Nullability.nonnull
+            }.map { param, _ in
+                JavaIR.ifBlock(condition: "this." + Languages.java.snakeCaseToPropertyName(param) + " == null") { [
+                    "missingFields.add(\"" + Languages.java.snakeCaseToPropertyName(param) + "\");",
+                ] }
+            } + [JavaIR.ifBlock(condition: "missingFields.size() > 0") { [
+                "throw new IllegalArgumentException(\"Model \(className) cannot be constructed with missing fields: \" + StringUtils.join(missingFields, \",\"));",
```

Comment:
Can we use something other than `StringUtils` to build this string? It would be nicer not to introduce a dependency on `org.apache.commons`. (This would fix the current CI build failure, too.)

Can we use Java 8's `String.join()` instead?

Reply:
I can't use Java 8 methods, but I'll just build up the string manually.

Reply:
I played with this a little bit, and I think we could get away with just using `missingFields.toString()`:

```java
List<String> missingFields = new ArrayList<>();
missingFields.add("one");
missingFields.add("two");
System.out.println(missingFields.toString());
// Output: [one, two]
```

That would simplify this bit of code a good amount. What do you think?

Reply:
Good call! Done.